torch/multiprocessing/reductions.py

```diff
@@ -38,26 +38,42 @@ def reduce_event(event):
     return (rebuild_event, (event.ipc_handle(),))
 
 
-def rebuild_tensor(cls, storage, metadata):
+def rebuild_tensor(cls, storage, metadata, requires_grad):
     storage_offset, size, stride = metadata
-    return torch._utils._rebuild_tensor(storage, storage_offset, size, stride)
+    t = torch._utils._rebuild_tensor(storage, storage_offset, size, stride)
+    if requires_grad is not None:
```

Comment:
I might be missing something, but does this mean that if I have a tensor that requires grad, and it gets serialized in one process it will become a nn.Parameter?

Reply:
@fmassa Actually this `requires_grad !=None` only when `is_parameter=True`. So it should be fine, although it's a bit misleading in name. 
Although this means if we share a tensor with requires_grad=True, the shared tensor is not set to requires_grad=True. Not sure if this is expected behavior.

Reply:
If that's the case, I wouldn't say it's expected

Reply:
That would be if `requires_grad` is False. None means that it's not a parameter. (Do you still think it's confusing? We can test for parameter-ness more explicitly if you like)